test/unit/math/rev/fun/as_value_array_or_scalar_test.cpp

```diff
@@ -0,0 +1,73 @@
+#include <stan/math/rev.hpp>
+#include <test/unit/util.hpp>
+#include <gtest/gtest.h>
+#include <vector>
+
+TEST(MathFunRev, as_value_array_or_scalar_scalar) {
+  double b_val = 4;
+  stan::math::var b(b_val);
+  EXPECT_EQ(b_val, stan::math::as_value_array_or_scalar(b));
+}
+
+TEST(MathFunRev, as_value_array_or_scalar_std_vector_lvalue) {
+  int n = 100;
+  Eigen::ArrayXd a_val = Eigen::ArrayXd::Random(n);
+  std::vector<double> b_val(n);
+  for (int i = 0; i < n; i++) {
+    b_val[i] = a_val.coeff(i);
+  }
+  std::vector<stan::math::var> b(b_val.begin(), b_val.end());
+  const auto& tmp = stan::math::as_value_array_or_scalar(b);
+  Eigen::ArrayXd res = tmp;
+  EXPECT_MATRIX_EQ(res, a_val);
+}
+
+TEST(MathFunRev, as_value_array_or_scalar_std_vector_rvalue) {
+  int n = 100;
+  Eigen::ArrayXd a_val = Eigen::ArrayXd::Random(n);
+  std::vector<double> b_val(n);
+  for (int i = 0; i < n; i++) {
+    b_val[i] = a_val.coeff(i);
+  }
+  std::vector<stan::math::var> b(b_val.begin(), b_val.end());
+  const auto& tmp = stan::math::as_value_array_or_scalar(std::move(b));
+  Eigen::ArrayXd res = tmp;
```

Comment:
Not testing the move. Is it not necessary to test it or something? It was tested in prim. Same comment applies for the rev tests for `as_value_column_vector_or_scalar` and `as_value_column_array_or_scalar`.

Reply:
I think for rev it's not since value_of is called and a copy is made